Sources/Array+.swift

```diff
@@ -29,15 +29,42 @@ public extension Array where Element: Hashable {
 }
 
 public extension Collection where Element: Hashable {
+    /// Returns a collection with no duplicate elememts.
+    ///
+    /// NOTE: The elementa are not garanteed to be in the same order as the original colection.
+    ///
+    /// Complexity: O(n), where n is the number of elements in the collection.
     public func unified() -> [Element] {
+        var set = Set<Element>()
         return reduce(into: []) {
-            if !$0.contains($1) {
+            if set.insert($0).inserted {
                 $0.append($1)
             }
         }
     }
 }
 
+public extension RandomAccessCollection where Element: Hashable {
```

Comment:
`RandomAccessCollection` is `BidirectionalCollection` and also `Collection`.
So, I think that this is unnecessary code. Don't you think so? 🤔 

```swift
public protocol RandomAccessCollection : BidirectionalCollection where Self.Indices : RandomAccessCollection, Self.SubSequence : RandomAccessCollection { }
public protocol BidirectionalCollection : Collection where Self.Indices : BidirectionalCollection, Self.SubSequence : BidirectionalCollection { }
```

Reply:
Sorry, I meant RangeReplacable